tests/core/test_integration/test_detection_structure_splitting.py

```diff
@@ -46,3 +50,34 @@ def test_structure_splitting(signal_array, background_array):
         voxel_sizes,
         n_free_cpus=0,
     )
+
+
+def test_underflow_issue_435():
+    # two cells centered at (9, 10, 10), (19, 10, 10) with radius 5
+    p1 = np.array([9, 10, 10])
+    p2 = np.array([19, 10, 10])
+    radius = 5
+
+    bright_voxels = np.zeros((20, 20, 30), dtype=np.bool_)
+
+    pos = np.empty((20, 20, 30, 3))
+    pos[:, :, :, 0] = np.arange(20).reshape((-1, 1, 1))
+    pos[:, :, :, 1] = np.arange(20).reshape((1, -1, 1))
+    pos[:, :, :, 2] = np.arange(30).reshape((1, 1, -1))
```

Comment:
```suggestion
    bright_voxels = np.zeros((30, 20, 20), dtype=np.bool_)

    pos = np.empty((30, 20, 20, 3))
    pos[:, :, :, 0] = np.arange(30).reshape((-1, 1, 1))
    pos[:, :, :, 1] = np.arange(20).reshape((1, -1, 1))
    pos[:, :, :, 2] = np.arange(20).reshape((1, 1, -1))
```


Did we want to have both cells entirely in the array? If we do, I think we need to reverse the axis dimensions here

Reply:
phew 🤦